File: src/main/java/com/saomc/util/PlayerStats.java
package com.saomc.util;

import com.saomc.api.info.IPlayerStatsProvider;

/**
 * Part of saoui
 *
 * @author Bluexin
 */
public class PlayerStats {
    private static PlayerStats instance;
    private final IPlayerStatsProvider stats;

    private PlayerStats(IPlayerStatsProvider stats) {
        this.stats = stats;
    }

    public static void init(IPlayerStatsProvider provider) {
        if (instance != null) throw new IllegalStateException("PlayerStats already initialized!");
        instance = new PlayerStats(provider);
    }

    public static PlayerStats instance() {
        if (instance == null) throw new IllegalStateException("PlayerStats not initialized!");
        return instance;
    }

    public IPlayerStatsProvider getStats() {
        return this.stats;
    }
}

File: src/main/java/com/saomc/saoui/util/PlayerStats.kt
package com.saomc.saoui.util

import com.saomc.saoui.api.info.IPlayerStatsProvider

/**
 * Part of saoui

 * @author Bluexin
 */
class PlayerStats private constructor(val stats: IPlayerStatsProvider) {
    companion object {
        private var instance: PlayerStats? = null

        fun init(provider: IPlayerStatsProvider) {
            if (instance != null) throw IllegalStateException("PlayerStats already initialized!")
            instance = PlayerStats(provider)
        }

        fun instance(): PlayerStats {
            if (instance == null) throw IllegalStateException("PlayerStats not initialized!")
            return instance!!
        }
    }
}
